From 2984765a7aa14b644e25f986c4cecf79879342e3 Mon Sep 17 00:00:00 2001
From: Stanley.Miao <stanley.miao@windriver.com>
Date: Thu, 17 Jun 2010 20:11:21 -0700
Subject: [PATCH] lttng: fix omap trace clock

export the clock symbols from the newly created clkt34xx_dpll3m2.

Signed-off-by: Stanley.Miao <stanley.miao@windriver.com>
---
 arch/arm/mach-omap2/clkt34xx_dpll3m2.c |    5 +++++
 arch/arm/mach-omap2/clock34xx.c        |    2 --
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/clkt34xx_dpll3m2.c b/arch/arm/mach-omap2/clkt34xx_dpll3m2.c
index b2b1e37..0c838cf 100644
--- a/arch/arm/mach-omap2/clkt34xx_dpll3m2.c
+++ b/arch/arm/mach-omap2/clkt34xx_dpll3m2.c
@@ -32,6 +32,9 @@
 
 #define CYCLES_PER_MHZ			1000000
 
+unsigned long long cpu_hz;
+EXPORT_SYMBOL(cpu_hz);
+
 /*
  * CORE DPLL (DPLL3) M2 divider rate programming functions
  *
@@ -78,6 +81,8 @@ int omap3_core_dpll_m2_set_rate(struct clk *clk, unsigned long rate)
 		pr_debug("clock: will unlock SDRC DLL\n");
 		unlock_dll = 1;
 	}
+	
+	cpu_hz = arm_fck_p->rate;
 
 	/*
 	 * XXX This only needs to be done when the CPU frequency changes
diff --git a/arch/arm/mach-omap2/clock34xx.c b/arch/arm/mach-omap2/clock34xx.c
index 371da0a..a12103a 100644
--- a/arch/arm/mach-omap2/clock34xx.c
+++ b/arch/arm/mach-omap2/clock34xx.c
@@ -125,6 +125,4 @@ const struct clkops clkops_omap3430es2_hsotgusb_wait = {
 	.find_companion = omap2_clk_dflt_find_companion,
 };
 
-unsigned long long cpu_hz;
-EXPORT_SYMBOL(cpu_hz);
 
-- 
1.6.5.2

